url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/file/TestForVirtualFile.java

```diff
@@ -0,0 +1,319 @@
+package net.hetimatan.io.file;
+
+import java.io.File;
+import java.io.IOException;
+
+
+
+import net.hetimatan.io.filen.CashKyoroFile;
+import junit.framework.TestCase;
+public class TestForVirtualFile extends TestCase {
+
+	public void log(String message) {
+//		System.out.println("kiyo:"+"TestForViertualFile:"+message);
+	}
+
+	public void testHello() {
+		log("testHello");
+	}
+
+	public void testZeroByte() throws IOException {
+		File testPath = new File(getFile(),"__001__.txt");
+		CashKyoroFile vf = null;
+		try {
+			testPath.delete();
+			testPath.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)